### PR TITLE
Docs base URL remaster

### PR DESCRIPTION
**Pull request by prrao87**
## Purpose of PR

The current base URL of the Kùzu docs aren't very Google search-friendly. When you type "Kuzu docs" in google, the documentation page basically doesn't even show up, because the base URL is set as `docusaurus`. My take is that when Google indexes pages, it looks for terms like "documentation" or "docs" in the metadata of the page header or the URL itself. In our case, it's `docusaurus`, so the google crawler doesn't seem to index the entire documentation website correctly.

The following is what you obtain (more or less) when you type "Kuzu docs" into Google search. As can be seen, **none** of the top 6 results are useful, as none of them leads you directly to the documentation, which is what you'd want as a user.

image

## Fix

The base URL should be `docs`, as this gives a much more intuitive url that google can hopefully index better: `https://kuzudb.com/docs`.

## How has this been tested?

I went through all the documents where we make a direct URL reference to `https://kuzudb.com/docusaurus` and replaced it with relative URLs `/docs` instead.

On my local server, I checked that all mouse-clicks on the relative URLs lead back to the same documentation pages, with no broken links to the best of my knowledge.

I think this would help improve SEO and lead users much more easily to our documentation page once they start using Kùzu.

### CODE REVIEW

**Comment by semihsalihoglu-uw** [comment]
I'll let @mewim review this.

**Review by ray6080 — APPROVED** [review]
I tried locally and this works fine. Maybe @mewim wants to take a quick look?

**Comment by prrao87** [comment]
Yup, the build workflow would need to be updated once it's merged to reflect the new paths too. Looking forward to the review from @mewim as well.

**Comment by mewim** [comment]
I would not change the base url for now. Currently we have many social media posts with links to blogs and docs starting with the `/docusaurus` base URL. While I realize that this is not a good choice, changing the base URL will make all these links dead and it is hard or impossible to retroactively update all of these social media posts. 

Alternatively, we can set up a redirection for all the URLs to make the social media posts work. We had a set up for the original blog posts under the old Just the Docs template [here](https://github.com/kuzudb/kuzudb.github.io/tree/main/blog). 

However, in longer term, we have a plan to also migrate the current home page to docusaurus. Unifying the home page under docusaurus provides the following advantages:
- Sharing some code and UI components such as the navbar.
- Enabling some more advanced frontend features powered by React.js.
- For search engine optimization, docusaurus generates a [site map](https://github.com/kuzudb/kuzudb.github.io/blob/main/docusaurus/sitemap.xml) but it is not picked up by search engines right now due to hosting docusaurus as a sub-site. If we host the entire website using docusaurus, the site map will be exposed and should improve search performance.
- Simplify the build process.

But with the home page migrations, the base URL for blogs will change again (from `/docs/blog` to `/blog`), which means we will have to maintain three different versions of redirections. This can be quite troublesome. I would suggest that we migrate the homepage first, so the `docusaurus` prefix will be naturally eliminated and we are left with only `/blog` and `/docs`. Then, we set up one redirection from `/docusaurus/blog` to `/blog`.

**Review by mewim — CHANGES_REQUESTED** [review]
See my comment

**Comment by prrao87** [comment]
@mewim that plan makes a lot of sense. I also wanted the blog page to point to /blog and not /docs/blog but wasn't sure how that could be done.

Do you think this PR can be kept open (as a draft) till we migrate the homepage? And the other PR in the homepage repo can be closed because we would be archiving that repo in any case.

And once the homepage migration is complete, we could then revisit this draft and update the blog URLs as /blog.

**Comment by mewim** [comment]
Yeah we can keep this as a draft for a while. We can revisit this after we migrate the home page.

**Comment by mewim** [comment]
We have completely separate the blog site from the docs site

**Comment by prrao87** [comment]
@mewim we still need to change the root directory from `docusaurus` to `docs` and set up redirects, correct? It's still tedious to find the Kuzu docs via a google search (typing "kuzu docs" never leads us to the docusaurus home page).

**Comment by mewim** [comment]
The docs are now at https://docs.kuzudb.com

**Comment by prrao87** [comment]
That's great!

**Pull request closed without merging**
